src/basic/hostname-util.c

```diff
@@ -196,8 +196,13 @@ bool is_gateway_hostname(const char *hostname) {
          * synthetic "gateway" host. */
 
         return
-                strcaseeq(hostname, "gateway") ||
-                strcaseeq(hostname, "gateway.");
+#if ENABLE_GATEWAY_HOSTNAME
+                strcaseeq(hostname, "gateway") || strcaseeq(hostname, "gateway.") ||
+#endif
+#if ENABLE__GATEWAY_HOSTNAME
+                strcaseeq(hostname, "_gateway") || strcaseeq(hostname, "_gateway.") ||
+#endif
```

Comment:
i don't think this second `#if` should be there. Let's check for the new name unconditionally, and only conditionalize the old name.